### PR TITLE
Use local variable for match_input_once to avoid saving it in checkpoint

### LINKED ISSUE #6786: Variable "matching_filenames" @  tf.train.match_filenames_once() should be local?
This var holds the matched file names for the input stream pipeline.
Since it is added to the graph, when using the contractor of tf.train.Saver() it is saved
as part of the checkpoint state.
Hence loading saved models from checkpoints for inference, the training file names are loaded instead of the val / inference files.

Shouldn't this Var become a local variable? will it not be saved by default treated as local?

### PR DESCRIPTION
**Pull request by yaroslavvb**
Fixes #6786

### CODE REVIEW

**Comment by josh11b** [comment]
Are we certain that users don't want the additional reproducibilty of saving this list of filenames in the checkpoint?  If it isn't saved, you might run this again and get a different answer and be surprised that suddenly your training is behaving differently.

**Comment by yaroslavvb** [comment]
In the original issue, the surprise was you run `tf.train.match_filenames_once` was run to generate list of testing filenames. Then pre-trained model was loaded from checkpoint, which overwrote this list with training filenames. 100% test set accuracy, woohoo!

Saved filenames are useful to restore while training a model, but are not useful to restore when loading this model for testing. It seems there needs to be some kind of distinction between model parameters and checkpointed things that are not model parameters.

**Comment by josh11b** [comment]
Jenkins, test this please.

**Review by josh11b — APPROVED** [review]

**Comment by yaroslavvb** [comment]
The failure is due to timeout
```
//tensorflow/contrib/layers:feature_column_ops_test                     TIMEOUT in 65.0s
  /var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/testlogs/tensorflow/contrib/layers/feature_column_ops_test/test.log
```

**Comment by yaroslavvb** [comment]
btw, unrelated timeout issue has been fixed

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Comment by rmlarsen** [comment]
@yaroslavvb I'm afraid that this broke an internal test. I'll be reverting this in the next push, while I investigate.

**Comment by yaroslavvb** [comment]
@rmlarsen my first guess would be that the test is trying to restore "matching_filenames" variable from checkpoint

**Comment by rmlarsen** [comment]
@yaroslavvb thanks for the hint. Will take a look later.
